spec/integration/util/windows/registry_spec.rb

```diff
@@ -163,5 +163,75 @@ def expects_registry_value(array)
         end
       end
     end
+
+    context "when reading binary values" do
+      let (:hklm) { Win32::Registry::HKEY_LOCAL_MACHINE }
+      let (:puppet_key) { "SOFTWARE\\Puppet Labs"}
+      let (:subkey_name) { "PuppetRegistryTest" }
+      let (:guid) { SecureRandom.uuid }
+
+      before :each do
+        hklm.create("#{puppet_key}\\#{subkey_name}", Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          reg.write("#{guid}", Win32::Registry::REG_BINARY, "abinarystring")
+        end
+      end
+
+      after(:each) do
+        hklm.open(puppet_key, Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          subject.delete_key(reg, subkey_name)
+        end
+      end
+
+      it "should return binary values" do
+
+        hklm.open("#{puppet_key}\\#{subkey_name}", Win32::Registry::KEY_ALL_ACCESS) do |reg|
+          vals = subject.values(reg)
+
+          expect(vals).to have_key(guid)
+          written = vals[guid]
+          expect(written).to eq("abinarystring")
+        end
+
+      end
+    end
+
+    context "when reading corrupt values" do
+      let (:hklm) { Win32::Registry::HKEY_LOCAL_MACHINE }
+      let (:puppet_key) { "SOFTWARE\\Puppet Labs"}
+      let (:subkey_name) { "PuppetRegistryTest" }
+      let (:guid) { SecureRandom.uuid }
+
+      before :each do
+        # Create a registry import file with a corrupt DWORD ( < 4 bytes in length)
+        tempfilename = "#{ENV['temp']}\\BadDword-PuppetRegistryTest.Reg"
```

Comment:
I will look at using the helper class instead of the current method.

I tried creating invalid values using API calls, WMI and REG.EXE and they all protected me from making a bad call.  REGEDIT.EXE was the only method that I could find that would allow me to inject a bad DWORD.  I do not believe regedit will accept STDIN so the only method left was a crafted temporary reg file to import.